datasets/hlgd/hlgd.py

```diff
@@ -0,0 +1,115 @@
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "timeline_id": datasets.features.ClassLabel(names=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+                "headline_a": datasets.Value("string"),
+                "headline_b": datasets.Value("string"),
+                "date_a": datasets.Value("string"),
+                "date_b": datasets.Value("string"),
+                "url_a": datasets.Value("string"),
+                "url_b": datasets.Value("string"),
+                "label": datasets.features.ClassLabel(names=[0, 1]),
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+
+        data_dir = dl_manager.download_and_extract(_DOWNLOAD_URL)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train.json"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, "test.json"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "dev.json"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self, filepath, split  # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    ):
+        """Yields examples as (key, example) tuples."""
+        # This method handles input defined in _split_generators to yield (key, example) tuples from the dataset.
+        # The `key` is here for legacy reason (tfds) and is not important in itself.
+
+        with open(filepath, "r") as f:
+            dataset_split = json.load(f)
+        return enumerate(dataset_split)
```

Comment:
Here also you need to yield the dataset values with corresponding features. This might help:
```
for id_, row in enumerate(dataset_split):
      yield id_, {
          "timeline_id": row['timeline_id'],
          "headline_a": row['headline_a'],
          "headline_b": row['headline_b'],
          "date_a": row['date_a'],
          "date_b": row['date_b'],
          "url_a": row['url_a'],
          "url_b": row['url_b'],
          "label": row['label'],
      }
```

Reply:
Got it I've added this!